autoconnect/autoconnect-ws/src/error.rs

```diff
@@ -61,14 +61,18 @@ impl ReportableError for WSError {
     fn is_sentry_event(&self) -> bool {
         match &self.kind {
             WSErrorKind::SM(e) => e.is_sentry_event(),
-            e => !matches!(e, WSErrorKind::Json(_) | WSErrorKind::SessionClosed(_)),
+            e => !matches!(
+                e,
+                WSErrorKind::PongTimeout | WSErrorKind::Json(_) | WSErrorKind::SessionClosed(_)
+            ),
         }
     }
 
     fn metric_label(&self) -> Option<&'static str> {
         match &self.kind {
             WSErrorKind::SM(e) => e.metric_label(),
             // TODO:
+            WSErrorKind::PongTimeout => Some("pong_timeout"),
```

Comment:
Let's not even metric this for a couple reasons: legacy autopush didn't, they're not terribly important and we're collecting these error events in the session's single `info!` log message, where its`disconnect_reason` field would be set to "Timeout waiting for Pong".

Apparently this is even more verbose than legacy autopush which treats a number of common errors like `PongTimeout` as being no Error, leading to an empty string for `disconnect_reason`: https://github.com/mozilla-services/autopush-rs/blob/f8e6525/autopush/src/client.rs#L583

Maybe autoconnect should even eventually do the same? Basically consider timeouts/io errors and the like as not really errors for a long lived websocket connection.

```suggestion
```

Reply:
I wasn't sure if there were more errors we should block, but it looks like that's the only one so far.

Reply:
Looking at this, I'm thinking that we should match rather than exclude, so I am saying it's a sentry event if we match:

WSErrorKind::SM(_) 
| WSErrorKind::Protocol(_)
| WSErrorKind::RegistryDisconnected

We're ignoring everything else. 

Sound good?